outlook/src/classes/Company.ts

```diff
@@ -33,10 +42,27 @@ class Company {
         this.additionalInfo = {};
     }
 
-    static fromJSON(o: Object): Company {
+    static fromJSON(o: Object, enrichmentInfo?: EnrichmentInfo): Company {
         if (!o) return new Company();
         const company = Object.assign(new Company(), o);
         company.address = Address.fromJSON(o['address']);
+
+        if (enrichmentInfo != null) {
+            if (enrichmentInfo.type == EnrichmentInfoType.NoData) {
+                company.enrichmentStatus = EnrichmentStatus.enrichmentEmpty;
+            } else {
+                if (EnrichmentInfoType.CompanyCreated || EnrichmentInfoType.CompanyUpdated) {
+                    company.enrichmentStatus = EnrichmentStatus.enriched;
+                }
+            }
+        } else {
+            if (company.hasInsights()) {
```

Comment:
Since "hasInsights" is a bit redundant with "enrichmentStatus" now, I suggest you inline the code here and remove the method (from what I have seen it's only used once).

Reply:
True, it has now been inlined.